clab/docker.go

```diff
@@ -179,6 +187,45 @@ func (c *cLab) CreateContainer(ctx context.Context, node *Node) (err error) {
 	return createContainerNS(node.Pid, node.LongName)
 }
 
+func (c *cLab) PullImageIfRequired(ctx context.Context, node *Node) (err error) {
+	filter := filters.NewArgs()
+	filter.Add("reference", node.Image)
+
+	ilo := types.ImageListOptions{
+		All:     false,
+		Filters: filter,
+	}
+
+	log.Debugf("Looking up %s Docker image", node.Image)
+
+	images, err := c.DockerClient.ImageList(ctx, ilo)
+	_ = images
+	// Image doesn't exist, need to pull it
+	if len(images) <= 0 {
+		// might need canonical name e.g.
+		//    -> alpine == docker.io/library/alpine
+		//    -> foo/bar == docker.io/foo/bar
+		//    -> docker.elastic.co/elasticsearch/elasticsearch == docker.elastic.co/elasticsearch/elasticsearch
+		canonicalImageName := node.Image
+		if strings.Count(node.Image, "/") == 0 {
+			canonicalImageName = "docker.io/library/" + node.Image
+		} else if strings.Count(node.Image, "/") == 1 {
+			canonicalImageName = "docker.io/" + node.Image
+		}
+		log.Infof("Pulling %s Docker image", node.Image)
+		reader, err := c.DockerClient.ImagePull(ctx, canonicalImageName, types.ImagePullOptions{})
+		// must read from reader, otherwise image is not properly pulled
+		io.Copy(ioutil.Discard, reader)
+		log.Infof("Done pulling %s", canonicalImageName)
+		if err != nil {
+			panic(err)
```

Comment:
lets return an error here instead of panicing

Reply:
will do